components/places/android/src/main/java/org/mozilla/places/PlacesConnection.kt

```diff
@@ -233,13 +250,44 @@ interface PlacesAPI {
      *
      */
     fun sync(syncInfo: SyncAuthInfo)
+
+    /**
+     * Interrupt ongoing operations running on a separate thread.
+     */
+    fun interrupt()
 }
 
+internal class InterruptHandle internal constructor(raw: RawPlacesInterruptHandle): AutoCloseable {
+    // We synchronize all accesses, so this probably doesn't need AtomicReference.
```

Comment:
I think that's a question for @grigoryk! 😄

Reply:
This is consistent with how we've done it in the past and most other usage, so lets keep it this way.